.github/workflows/link-check.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and check Links
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build_and_serve:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install Dependencies
+        run: yarn install
+
+      - name: Build Next.js App
+        run: yarn build
+
+      - name: Serve App Locally
+        run: yarn start &
+
+      - name: Wait for App to Start
+        run: sleep 20
+
+  linkChecker:
+    runs-on: ubuntu-latest
+    needs: build_and_serve
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Link Checker
+        uses: lycheeverse/lychee-action@v1.9.0
```

Comment:
If you always want to use the latest features and avoid breaking changes, you can replace the version with `lycheeverse/lychee-action@v1`.

Reference: https://github.com/lycheeverse/lychee-action - checkout README.md Usage Section